### PR TITLE
Stanley param in MOM_isopycnal_slopes

### PR DESCRIPTION
**Pull request by jskenigson**

### CODE REVIEW

File: src/core/MOM_isopycnal_slopes.F90 [hunk 1]
```diff
@@ -202,6 +215,19 @@ subroutine calc_isoneutral_slopes(G, GV, US, h, e, tv, dt_kappa_smooth, &
...
+      do i=is-1,ie+1
+        pres_h(i) = pres(i,j,K)
+        T_h(i) = 0.5*(T(i,j,k) + T(i,j,k-1))
+        S_h(i) = 0.5*(S(i,j,k) + S(i,j,k-1))
+      enddo
+      ! The second line below would correspond to arguments
+      !            drho_dS_dS, drho_dS_dT, drho_dT_dT, drho_dS_dP, drho_dT_dP, &
+      call calculate_density_second_derivs(T_h, S_h, pres_h, &
+                   scrap, scrap, drho_dT_dT_h, scrap, scrap, &
+                   is-1, ie-is+3, tv%eqn_of_state)
```
Comment (iangrooms) [inline src/core/MOM_isopycnal_slopes.F90:228]: Why is it `ie-is+3` here when further down it's `ie-is+1`?

Comment (jskenigson) [reply src/core/MOM_isopycnal_slopes.F90:228]: The loops that calculate drdiA, drdjA, etc. operate over different arrays of indices. This calculation should work for indices is-1 to ie+1, i.e., over ie-is+3 points, while the one below is for is to ie, over ie-is+1 points.

File: src/parameterizations/lateral/MOM_thickness_diffuse.F90 [hunk 2]
```diff
@@ -804,8 +805,8 @@ subroutine thickness_diffuse_full(h, e, Kh_u, Kh_v, tv, uhD, vhD, cg1, dt, G, GV
         call calculate_density_derivs(T_u, S_u, pres_u, drho_dT_u, drho_dS_u, &
                                       tv%eqn_of_state, EOSdom_u)
       endif
-      if (use_Stanley) then
-        do i=is-1,ie
+      if (use_stanley) then
+        do i=is-1,ie+1
```
Comment (iangrooms) [inline src/parameterizations/lateral/MOM_thickness_diffuse.F90:809]: I assume the `+1` here and the `+2` becoming ~+3` on line 818 are the bug you mentioned in the meeting?

Comment (jskenigson) [reply src/parameterizations/lateral/MOM_thickness_diffuse.F90:809]: Yes, correct.

**Review by iangrooms — COMMENTED** [review]
I left a couple of questions about indexing loops.

File: src/core/MOM_isopycnal_slopes.F90 (hunk 1 above)
Comment (iangrooms) [reply src/core/MOM_isopycnal_slopes.F90:228]: OK, makes sense.

**Review by iangrooms — APPROVED** [review]
OK, looks good to me

**Pull request merged**